plumber/plumber-android-core/src/main/java/leakcanary/AndroidLeakFixes.kt

```diff
@@ -707,6 +707,26 @@ enum class AndroidLeakFixes {
         SharkLog.d(ignored) { "Unable to fix SpellChecker leak" }
       }
     }
+  },
+
+  /**
+   * PermissionControllerManager stores the first context it's initialized with forever.
+   * Sometimes it's an Activity context which then leaks after Activity is destroyed.
+   *
+   * This fix makes sure the PermissionControllerManager is created with the application context.
+   *
+   * For Pixel devices the issue can be tracked here
+   * https://issuetracker.google.com/issues/318415056
+   */
+  PERMISSION_CONTROLLER_MANAGER {
+    @SuppressLint("WrongConstant")
+    override fun apply(application: Application) {
+      try {
+        application.getSystemService("permission_controller")
```

Comment:
Should this only happen after a certain API level?

In your example repro on the google issue, you called `revokeSelfPermissionsOnKill` which is API 33+

Reply:
The `"permission_controller"` string was introduced here: https://cs.android.com/android/_/android/platform/frameworks/base/+/bc054d8436709b1396d242bcbbf2a9285b7a2575

That string doesn't exist in android-9.0.0_r61 : https://cs.android.com/android/platform/superproject/+/android-9.0.0_r61:frameworks/base/core/java/android/content/Context.java

The string does exist in android-10.0.0_r1 https://cs.android.com/android/platform/superproject/+/android-10.0.0_r1:frameworks/base/core/java/android/content/Context.java;l=4140;drc=5016a7891a8c5fe1c8ca5c24e83595b7973fc6b0

So that'd be API 29+